### ISSUE TITLE
What to do for torch result_type()?

### ISSUE THREAD

**Issue by tylerjereddy** [data-apis/array-api-compat#43]
While working on https://github.com/scipy/scipy/issues/18286 and https://github.com/tylerjereddy/scipy/pull/70 to test passing through different array types in `scipy.signal.welch()` I ran into the same thing Ralf did a few years ago here: https://github.com/pytorch/pytorch/issues/51284

Since that has been open for quite some time, should we add a shim to deal with `torch.result_type` only accepting numbers/tensors and not types proper? Or should downstream libs deal with this in their own way?

Traceback I saw was:

```
/home/tyler/python_310_scipy_dev_work/lib/python3.10/site-packages/array_api_compat-1.2-py3.10.egg/array_api_compat/torch/_aliases.py:124: in result_type
    return torch.result_type(x, y)
E   TypeError: result_type() received an invalid combination of arguments - got (Tensor, torch.dtype), but expected one of:
E    * (Tensor tensor, Tensor other)
E         didn't match because some of the arguments have invalid types: (Tensor, !torch.dtype!)
E    * (Number scalar, Tensor tensor)
E         didn't match because some of the arguments have invalid types: (!Tensor!, !torch.dtype!)
E    * (Tensor tensor, Number other)
E         didn't match because some of the arguments have invalid types: (Tensor, !torch.dtype!)
E    * (Number scalar1, Number scalar2)
E         didn't match because some of the arguments have invalid types: (!Tensor!, !torch.dtype!)
        arrays_and_dtypes = (tensor([0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.]), torch.complex64)
        x          = tensor([0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.])
        xdt        = torch.float32
        y          = torch.complex64
        ydt        = torch.complex64
```

**Comment by asmeurer** [comment]
This will be fixed by #38. We just pass through to torch.result_type for non-array API dtypes but fix it for array API dtypes. But complex dtype support was only added in that PR.

**Comment by rgommers** [comment]
gh-38 was merged, so I'll close this issue.
